tls/s2n_signature_algorithms.c

```diff
@@ -50,13 +52,25 @@ static int s2n_signature_scheme_valid_to_offer(struct s2n_connection *conn, cons
 static int s2n_signature_scheme_valid_to_accept(struct s2n_connection *conn, const struct s2n_signature_scheme *scheme)
 {
     POSIX_ENSURE_REF(scheme);
+    POSIX_ENSURE_REF(conn);
 
     POSIX_GUARD(s2n_signature_scheme_valid_to_offer(conn, scheme));
 
     if (scheme->maximum_protocol_version != S2N_UNKNOWN_PROTOCOL_VERSION) {
         POSIX_ENSURE_LTE(conn->actual_protocol_version, scheme->maximum_protocol_version);
     }
 
+    POSIX_ENSURE_NE(conn->actual_protocol_version, S2N_UNKNOWN_PROTOCOL_VERSION);
+
+    if (conn->actual_protocol_version >= S2N_TLS13) {
+        POSIX_ENSURE_NE(scheme->hash_alg, S2N_HASH_SHA1);
+        POSIX_ENSURE_NE(scheme->sig_alg, S2N_SIGNATURE_RSA);
+    }
+
+    if (conn->actual_protocol_version < S2N_TLS13) {
+        POSIX_ENSURE_NE(scheme->sig_alg, S2N_SIGNATURE_RSA_PSS_PSS);
+    }
```

Comment:
Can't this now just be an else?

Reply:
Yes. I didn't change it because I thought it would be more clear. --> But looking at it after the change it seems more obviously correct; less prone to being broken by future changes.